app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt

```diff
@@ -301,30 +306,59 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
     }
   }
 
+  private fun requestExternalStoragePermission() {
+    ActivityCompat.requestPermissions(
+      requireActivity(), arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
+      REQUEST_STORAGE_PERMISSION
+    )
+  }
+
+  override fun onRequestPermissionsResult(
+    requestCode: Int,
+    permissions: Array<out String>,
+    grantResults: IntArray
+  ) {
+    super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+    if (requestCode == REQUEST_STORAGE_PERMISSION &&
+      grantResults.isNotEmpty() &&
+      grantResults[0] != PERMISSION_GRANTED
+    ) {
```

Comment:
We should ask here like this  

``` java
if (requestCode == REQUEST_STORAGE_PERMISSION && permissions.isNotEmpty() && permissions[0] != Manifest.permission.WRITE_EXTERNAL_STORAGE)

```
then at this point, the user doesn’t want to give us permission. Then we should do like this
``` Java
if (shouldShowRationale(WRITE_EXTERNAL_STORAGE)) {
          alertDialogShower.show(
            KiwixDialog.StoragePermissionRationale,
            ::requestStoragePermissionPermission
          )
  }

````

Reply:
done